pykafka/balancedconsumer.py

```diff
@@ -460,22 +473,16 @@ def _path_from_partition(self, p):
     def _remove_partitions(self, partitions):
         """Remove partitions from the zookeeper registry for this consumer.
 
-        Also remove these partitions from the consumer's internal
-        partition registry.
-
         :param partitions: The partitions to remove.
         :type partitions: Iterable of :class:`pykafka.partition.Partition`
         """
         for p in partitions:
-            assert p in self._partitions
+            # TODO pass zk node version to make sure we still own this node
```

Comment:
I like the idea of making pykafka znode-version-aware. What would we do here, though? I guess we'd raise an error if we found that we didn't own the node? Warn and pass? Trigger a new rebalance?
It's definitely not for this pull request, but it seems worthwhile to explore.